bm-persona/Libraries/LibraryDetailViewController.swift

```diff
@@ -0,0 +1,368 @@
+//  Copyright © 2020 RJ Pimentel. All rights reserved.
+//
+
+import UIKit
+import CoreLocation
+
+fileprivate let kHeaderFont: UIFont = Font.bold(24)
+fileprivate let kCardPadding: UIEdgeInsets = UIEdgeInsets(top: 5, left: 10, bottom: 5, right: 10)
+fileprivate let kViewMargin: CGFloat = 16
+
+class LibraryDetailViewController: UIViewController {
+
+    var scrollView: UIScrollView!
+    var library : Library!
+    var contentHelper: UIView!
+    var locationManager = CLLocationManager()
+    var userCoords = CLLocationCoordinate2D(latitude: 0.0 , longitude: 0.0 )
+    var distLabel = UILabel()
+    var overviewCard = CardView()
+    var hoursCard = CardView()
+    var bookButton = UIButton()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = .white
+        
+        //location stuff
+        locationManager.delegate = self
+        
+        view.layoutMargins = UIEdgeInsets(top: 16, left: 0, bottom: 0, right: 0)
+        setUpScrollView()
+        setUpOverviewCard()
+        setUpHoursCard()
+        //setUpTrafficCard()
+        setUpBookButton()
+        
+        
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager.requestLocation()
+        }
+    }
+}
+
+extension LibraryDetailViewController {
+    
+    func setUpOverviewCard() {
+        overviewCard.isUserInteractionEnabled = true
+        overviewCard.layoutMargins = kCardPadding
+        contentHelper.addSubview(overviewCard)
+        overviewCard.translatesAutoresizingMaskIntoConstraints = false
+        overviewCard.leftAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.leftAnchor).isActive = true
+        overviewCard.rightAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.rightAnchor).isActive = true
+        
+        let nameLabel = UILabel()
+        nameLabel.font = Font.bold(24)
+        nameLabel.text = library!.searchName
+        overviewCard.addSubview(nameLabel)
+        nameLabel.lineBreakMode = .byWordWrapping
+        nameLabel.numberOfLines = 0
+        nameLabel.translatesAutoresizingMaskIntoConstraints = false
+        nameLabel.topAnchor.constraint(equalTo: overviewCard.layoutMarginsGuide.topAnchor).isActive = true
+        nameLabel.leftAnchor.constraint(equalTo: overviewCard.layoutMarginsGuide.leftAnchor).isActive = true
+        nameLabel.widthAnchor.constraint(equalTo: overviewCard.layoutMarginsGuide.widthAnchor, multiplier: 0.5).isActive = true
+        nameLabel.adjustsFontSizeToFitWidth = true
+        
+        let libPic = UIImageView()
+        if library?.image != nil {
+            libPic.image = library?.image
+        } else {
+            DispatchQueue.global().async {
+                guard let imageData = try? Data(contentsOf: self.library!.imageURL!) else { return }
+                let image = UIImage(data: imageData)
+                DispatchQueue.main.async {
+                    libPic.image = image
```

Comment:
add library.image = image line here to save the loaded image so it can be displayed elsewhere/in the future